frontend/src/components/ModalSetUsername/ModalSetUsername.tsx

```diff
@@ -0,0 +1,50 @@
+import { Button, Dialog, DialogActions, DialogContent, TextField, Typography } from '@material-ui/core'
+import React from 'react'
+import useApplication from '../../shared/applications'
+
+export const ModalSetUsername: React.FC<{
```

Comment:
Could we call this component UsernameModal? It follows the pattern of <descriptorType><componentType> like LaunchButton or CopyButton
```suggestion
export const UsernameModal: React.FC<{
```